dashboard/pages/screening/callbacks.py

```diff
@@ -345,6 +371,52 @@ def on_z_score_range_update(n_clicks, figure, range):
     return new_figure
 
 
+# === STAGE 6 ===
+
+
+def on_save_results_click(
+    n_clicks: int, stored_uuid: str, filename: str, file_storage: FileStorage
+):
+    """
+    Callback for the save results button
+
+    :param n_clicks: number of clicks
+    :param stored_uuid: uuid of the stored data
+    :param filename: filename to save the results to
+    :param file_storage: storage object
+    :return: None
+    """
+
+    filename = filename + ".csv"
+    filestorage_filename = f"{stored_uuid}_echo_bmg_combined.csv"
+
+    if file_storage.file_exists(filestorage_filename):
+        return dcc.send_file(
+            file_storage.data_folder / filestorage_filename, filename=filename
+        )
+
+    echo_bmg_combined_df = pd.read_parquet(
+        pa.BufferReader(
+            file_storage.read_file(f"{stored_uuid}_echo_bmg_combined_df.pq")
+        ),
+    )
+
+    echo_bmg_combined_df = reorder_bmg_echo_columns(echo_bmg_combined_df)
+    csv_buffer = echo_bmg_combined_df.to_csv(
```

Comment:
Why not send the csv directly to the user instead of saving and deleting it? This approach causes issues as our base FileStorage does not have a `data_folder`, so if we ever decide to use any other place for storing files this approach won't work.

Have you considered using the approach described [here](https://dash.plotly.com/dash-core-components/download#:~:text=DOWNLOAD%20TEXT-,Downloading%20a%20Dataframe%20as%20a%20CSV%20file,-For%20downloading%20dataframes)?

e.g.
```py
return dcc.send_data_frame(echo_bmg_combined_df.to_csv, filename)
```

This way you do not have to have this complex save/delete interaction with a separate thread

Reply:
Makes sense, will do and delete unnecessary filestorage functions